src/subset/error.rs

```diff
@@ -27,3 +27,14 @@ pub enum Error {
 
 /// A subset result.
 pub type Result<T> = result::Result<T, Error>;
+
+/// Subset does not actually have any messages defined, so there's no real FaultKind to define here
```

Comment:
That documentation is a bit weird. Maybe something like: "Faults that can be detected in `Subset`."
It's clear from the set of variants that these can only originate in `Broadcast` or `BinaryAgreement`.